connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -717,6 +750,15 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
                 if (started)
                     updateListener.onTaskConfigUpdate(updatedTasks);
+            } else if (record.key().startsWith(RESTART_PREFIX)) {
+                RestartRequest request = recordToRestartRequest(record, value);
+                if (request == null) {
+                    return;
+                }
+                // Only notify the listener if this backing store is already successfully started (having caught up the first time)
+                if (started) {
```

Comment:
Since we print the warning inside `recordToRestartRequest` we can probably avoid the early return with:
```suggestion
                // Only notify the listener if this backing store is already successfully started (having caught up the first time)
                if (request != null && started) {
```